python/pyxet/pyxet/file_system.py

```diff
@@ -138,181 +182,119 @@ def branch_info(self, url):
-            raise ValueError("URL not in recognized format.")
-        prefix = '/'.join(components[:2]) + '/' + url_path.branch
-        attr = _manager.stat(url_path.remote, url_path.branch, url_path.path)
+        url_path = parse_url(url, self.domain, expect_branch = True)
+        attr = self._manager.stat(url_path.remote(), url_path.branch, url_path.path)
+
         if attr is None:
             raise FileNotFoundError(f"File not found {url}")
-        return {"name": prefix + '/' + url_path.path,
+
+        return {"name": url_path.name(),
                 "size": attr.size,
                 "type": attr.ftype,
                 "last_modified": None if len(attr.last_modified) == 0 else attr.last_modified}
 
     def make_repo(self, dest_path, private=False, **kwargs):
-        dest = parse_url(dest_path, self.domain)
-        if dest.path != '':
-            raise ValueError("Expecting xet://user/repo for destination")
+        dest = parse_url(dest_path, self.domain, expect_branch = False)
         if self.is_repo(dest_path):
             raise ValueError(f"{dest_path} already exists")
 
-        domain = self.domain
-        domain_split = domain.split('://')
-        scheme = 'https'
-        if len(domain_split) == 2:
-            scheme = domain_split[0]
-            domain = domain_split[1]
-
-        # dest_path is of the form xet://user/repo
-        split = dest_path.split('://')[-1].split('/')
-        assert (len(split) == 2)
-        owner, repo = split[0], split[1]
-
-        query = json.dumps({'name': repo, 'owner': owner, 'private': private})
-        ret = json.loads(bytes(_manager.api_query(f"{scheme}://{domain}", "", "post", query)))
+        query = json.dumps({'name': dest.repo, 'owner': dest.user, 'private': private})
+        ret = json.loads(bytes(self._manager.api_query(dest.domain_url(), "", "post", query)))
         return ret
 
     def fork_repo(self, origin_path, dest_path, **kwargs):
-        origin = parse_url(origin_path, self.domain)
-        dest = parse_url(dest_path, self.domain)
-        if origin.path != '':
-            raise ValueError("Expecting xet://user/repo for fork origin")
-        if dest.path != '':
-            raise ValueError("Expecting xet://user/repo for destination")
+        origin = parse_url(origin_path, self.domain, expect_branch = False)
+        dest = parse_url(dest_path, self.domain, expect_branch = False)
         if not self.is_repo(origin_path):
             raise ValueError(f"{origin_path} is not a repo")
         if self.is_repo(dest_path):
             raise ValueError(f"{dest_path} already exists")
-        user_and_repo = list(filter(None, dest_path.split('://')[-1].split('/')))
-        if len(user_and_repo) != 2:
-            raise ValueError("Expecting xet://user/repo for destination")
-        # validate username
-        user, new_repo = user_and_repo[0], user_and_repo[1]
-        real_username = self.get_username()
-        if user != real_username:
```

Comment:
Not sure if this is needed -- is this just a fast fail test? Does the server return informative message when this is true?

Reply:
Ah, good catch.  Will add that back in, but I think it's a fail-fast thing. (And probably hard-to-grok error messages like a generic permission denied)